src/Network/Broadcast/OutboundQueue.hs

```diff
@@ -815,6 +816,13 @@ intFailure OutQ{..} threadRegistry@TR{} p sendExecTime err = do
 hasRecentFailure :: MonadIO m => OutboundQ msg nid buck -> nid -> m Bool
 hasRecentFailure OutQ{..} nid = liftIO $ Set.member nid <$> readMVar qFailures
 
+-- | Reset internal statistics about failed nodes
+--
+-- This is useful when we know for external reasons that nodes may be reachable
+-- again, allowing the outbound queue to enqueue messages to those nodes.
+clearRecentFailures :: MonadIO m => OutboundQ msg nid buck -> m ()
+clearRecentFailures OutQ{..} = applyMVar_ qFailures $ const Set.empty
```

Comment:
Should kill any threads which are waiting to delete an element from this set. If we don't, they could prematurely clear subsequent failures.